recipes-openxt/qemu-dm/files/0027-nic-link-state-propagation.patch

```diff
@@ -0,0 +1,246 @@
+Intial Commit: Unknown
+
+################################################################################
+REMOVAL 
+################################################################################
+With the current OpenXT guest network configuration, removing this patch will
+trigger routing issues in the guest.
+
+################################################################################
+UPSTREAM PLAN 
+################################################################################
+There is no plan to upstream this patch, it is OpenXT specific.
+
+################################################################################
+INTERNAL DEPENDENCIES 
+################################################################################
+None
+
+################################################################################
+PATCHES 
+################################################################################
+--- a/hw/xen.h
++++ b/hw/xen.h
+@@ -38,6 +38,9 @@ void xen_piix_pci_write_config_client(ui
+ void xen_hvm_inject_msi(uint64_t addr, uint32_t data);
+ void xen_cmos_set_s3_resume(void *opaque, int irq, int level);
+ 
++int xenstore_register_nic(NetClientState *nc);
++int xenstore_unregister_nic(NetClientState *nc);
++
+ qemu_irq *xen_interrupt_controller_init(void);
+ 
+ int xen_init(void);
+--- a/hw/xen_backend.c
++++ b/hw/xen_backend.c
+@@ -38,6 +38,7 @@
+ #include "char/char.h"
+ #include "qemu/log.h"
+ #include "xen_backend.h"
++#include "qmp-commands.h"
+ 
+ #include <xen/grant_table.h>
+ 
+@@ -589,6 +590,143 @@ static int xenstore_scan(const char *typ
+     return 0;
+ }
+ 
++/*
++ * OpenXT: Get type and id from \([a-zA-Z]\+\)\([0-9]\+\).
++ */
++static void xenstore_nic_parse_name(const char *name, char *type, char *id)
++{
++    if (name) {
++        while (isalpha(*name)) {
++            *(type++) = *(name++);
++        }
++        while (isdigit(*name)) {
++            *(id++) = *(name++);
++        }
++    }
++    *type = '\0';
++    *id = '\0';
++}
++
++/*
++ * OpenXT: Get network backend type and ID from NIC information table.
++ */
++static int xenstore_net_client_state_fetch_backend_info(NetClientState *ncs, char *type, char *id)
++{
++    size_t i;
++
++    for (i = 0; i < MAX_NICS; ++i) {
++        /* We assume there is only one NIC per hub. So, if a NIC is on our
++         * hub, it is our back-end. */
++        if (nd_table[i].netdev == ncs->peer) {
++            xenstore_nic_parse_name(nd_table[i].name, type, id);
++            return 0;
++        }
++    }
++    return -1;
++}
++
++/* Define constants for network devices XenStore data representation. */
++#define XENSTORE_NET_TYPE_LEN   5   /* Currently "vif"|"vwif" */
++#define XENSTORE_NET_ID_LEN     5
++/* OpenXT: Read the appropriate Base Register and check if we have to change
++ * the device status. */
++static void xenstore_update_nic(void *opaque)
++{
++    NetClientState *nc = opaque;
```

Comment:
Actually, I am more afraid of the xenstore_update() function parsing values from XenStore node and using them as pointer without check.
But again, debug assert?